### ISSUE TITLE
[BUG] Minversion parameter not working as expected in v1.4.3

### ISSUE THREAD

**Issue by nhaleft** [Haivision/srt#2008]
**Describe the bug**
A listener with the minversion successfully rejects incoming connections from old clients, but valid connections are still blocked from connecting by the old connection.

**To Reproduce**
1. Run a v1.4.3 listener with minversion v1.4.2
2. Run a caller on a version of SRT that uses HSv4 (such as the one used by VLC) trying to connect to the listener
3. Run another caller on version 1.4.2 or 1.4.3 trying to connect to the listener
4. Observe the v1.4.2/3 caller fail to connect to the listener
5. Kill the old caller process (from step 2)
6. Observe the v1.4.2/3 caller connect to the listener when the old caller process is killed

**Expected behavior**
My expectation is that the connection that doesn't meet minversion requirements does not block other incoming connections.

**Desktop (please provide the following information):**
 - OS: [e.g. Windows, Linux, macOS,...]
 - SRT Version / commit ID: v1.4.3

**Additional context**
This bug has only been tested in the case where the "old" SRT caller uses HSv4. I have not confirmed that the same behavior still occurs if the "old" caller uses HSv5.

I believe I may have found the cause of this behavior, or a piece of code related to it. In core.cpp, within the `CUDT::interpretSrtHandshake` function, this if-statement appears to blindly accept connections using old handshakes:
```
if (hs.m_iVersion &lt; HS_VERSION_SRT1)
                return true; // do nothing
```
When this if-statement is changed as follows:
```
if (hs.m_iVersion < HS_VERSION_SRT1)
    {
        m_RejectReason = SRT_REJ_PEER;
        LOGC(cnlog.Error, log << "HS VERSION < HS_VERSION_SRT1; rejected because can't enforce encryption.");
        return false;
    }
```
HSv4 connections are blindly dropped instead of accepted, and valid connections are not blocked from connecting to the listener. So, a possible solution to this issue may be checking the minversion option in this if-statement.

**Comment by ethouris** [comment]
Ok, I tried to test it and I have so far observed that there's indeed one problem: The 1.2.3 version client (with HSv4) connects to the listener despite not satisfying the minimum version 1.4.2. The 1.3.4 caller is rejected, the 1.4.2 caller is accepted. I'll take a look at this.

But then, you are talking about a kinda different problem and I'd like to understand it correctly. Is it so that:
1. You have a listener with minversion=1.4.2
2. You have one app that attempts to connect and it is less than 1.3.0 (as I verified, it would succeed)
3. Simultaneously, another application is attempting to connect, it may be of 1.4.2 or 1.4.3, all will fail to connect...

First, I need to know then if the same thing happens if you don't put the minversion limitation on the listener, as I need to eliminate the case of blocking a listener by an attempting caller. If this is about `srt-live-transmit` application, note that this application doesn't accept any connections once one connection has been made, and the HSv4 clients seem to (incorrectly, but still) succeed.

**Comment by nhaleft** [comment]
The fix you implemented in #2010 is exactly what I was looking for, thanks!

If you still need answers to your questions, here they are:

1. The scenario you described in points 1, 2, and 3 is correct.
2. Without minversion, the behavior is similar but not exactly the same. If the listener doesn't have minversion, a v1.2.3 caller will successfully connect and receive traffic.
3. This is indeed about `srt-live-transmit`

Thanks for addressing this issue so quickly!
